### PR TITLE
1268 gae versions

### LINKED ISSUE #1268: OAuth redirects. Different GAE versions.
Issue description
In order to have OAuth working on different versions, everytime we deploy we need to update the list of `Allowed redirects`  (and allowed domains).

It's currently quite time consuming and annoying to do that for every new version.

We should  store the GAE version while using the default app for the OAuth flow and then redirect to the right version if the authentication is successful.

Traceback or other info

* Djangae version: x.x.x / master@commit
* Django version: x.x.x
* App Engine SDK version: 1.9.xx
* Using Cloud SDK: yes/no

### PR DESCRIPTION
**Pull request by flea89**
Fixes #1268 

Summary of changes proposed in this Pull Request:
- Add configuration to specify OAUTH2 redirect host
- Store incoming app version in oauth state
- Redirect to the right appengine from oauth2callback

PR checklist:
- [ ] Updated relevant documentation
- [ ] Updated CHANGELOG.md 
- [ ] Added tests for my change

### CODE REVIEW

File: djangae/contrib/googleauth/views.py [hunk 1]
```diff
@@ -41,33 +46,57 @@
...
+        scopes = set(scopes).union(set(additional_scopes))
+
+    if with_scope:
+        kwargs['scope'] = sorted(scopes)
+
+    original_url = f"{request.scheme}://{request.META['HTTP_HOST']}"
+
+    # Use hardcoded uri for oauth flow to avoid having to set redirect_urls
+    # for every single new app version.
+    url = OAUTH2_REDIRECT_URL if OAUTH2_REDIRECT_URL else original_url
```
Comment (sleepyjames) [inline djangae/contrib/googleauth/views.py:68]:
```suggestion
    url = OAUTH2_REDIRECT_URL or original_url
```

**Review by sleepyjames — CHANGES_REQUESTED** [review]
Nice one. Can you document the use of `OAUTH2_REDIRECT_URL` too please.

File: djangae/contrib/googleauth/views.py [hunk 2]
```diff
@@ -41,33 +46,57 @@
 # the token was granted, not the time we process it
 _TOKEN_EXPIRATION_GUARD_TIME = 5
 
+OAUTH2_REDIRECT_URL = getattr(settings, 'OAUTH2_REDIRECT_URL', None)
```
[the remark this answers is not in the document]
Comment (flea89) [reply djangae/contrib/googleauth/views.py]: Moved, thanks!

File: djangae/contrib/googleauth/tests/test_oauth.py [hunk 3]
```diff
@@ -85,10 +86,12 @@ def test_redirect_to_authorization_url(self):
             response = self.client.get(response.url, HTTP_HOST=live_server_domain)
             # check OAuthSession has been called properly
             self.assertEqual(auth_url.calls[0].args[1], 'https://accounts.google.com/o/oauth2/v2/auth')
-            self.assertEqual(auth_url.calls[0].kwargs, {
+            self.assertTrue(auth_url.calls[0].kwargs.items() > {
```
[the remark this answers is not in the document]
Comment (flea89) [reply djangae/contrib/googleauth/tests/test_oauth.py]: sure

**Comment by flea89** [comment]
Thanks for the feedback @sleepyjames , actioned everything and pushed some tests as well.

**Comment by sleepyjames** [comment]
Great, thanks @flea89

File: docs/googleauth.md [hunk 4]
```diff
@@ -107,6 +107,21 @@ GOOGLEAUTH_CLIENT_SECRET = "..."
 
 You must generate these values in the Google Cloud Console.
 
+When you configure OAuth Consent Screen and Credentials on Google Cloud Platform you are 
+required to configure a list of Authorised domains and Oauth redirects.
```
Comment (sleepyjames) [inline docs/googleauth.md:111]:
```suggestion
required to configure a list of Authorised domains and OAuth redirects.
```

**Review by sleepyjames — APPROVED** [review]

**Pull request merged**